### PR TITLE
[RG-250] Show OpenSTA timing engine messages and timing report

### PR DESCRIPTION
**Pull request by TarasSydor**
> ### Motivate of the pull request
This PR is extending timing analysis parser with OpenSTA messages (RG-250).

> ### Describe the technical details
In case compiler has OpenSTA timing option set, parsing will be made based on different messages. Due to lack of information on OpenSTA reporting, there are some shortcuts taken (timing info):
- Timing info is only represented by 'wns, tns' values. FMAX was not found in given examples;
- We only taken the info from timing_analysis.rpt as agreed with Ravi.
STARS info is present in raptor.log. Not sure whether it will be added to timing_analysis.rpt in the future, but currently it's not there and we don't read it.

Next to OpenSTA, there is one more improvement added, that makes sure file is not marked as 'parsed' in case it's empty. In some cases on Raptor the log is initially empty and filled afterwards.
>
> #### What does this pull request change?
OpenSTA messages:
image

At this point timing report only contains 2 rows and a table:
image

[RG-250]: https://rapidsilicon.atlassian.net/browse/RG-250?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by KochynVolodymyr — APPROVED** [review]

**Review by ravic-rs — CHANGES_REQUESTED** [review]
Please pay attention to CI checks once PR is created. How did this build pass locally ?

**Comment by TarasSydor** [comment]
> Please pay attention to CI checks once PR is created. How did this build pass locally ?

It was pushed from Raptor. Apparently something went wrong while merging branches.
Will check it out on local FOEDAG and see what's wrong.

**Review by KochynVolodymyr — APPROVED** [review]

File: src/Main/Foedag.cpp [hunk 1]
```diff
@@ -383,7 +383,7 @@ bool Foedag::initBatch() {
   m_projectFileLoader->registerComponent(
       new ProjectManagerComponent{m_compiler->ProjManager()},
       ComponentId::ProjectManager);
-  auto taskM = new TaskManager;
+  auto taskM = new TaskManager(m_compiler);
```
Comment (ravic-rs) [inline src/Main/Foedag.cpp:386]: Why do we need this change ?
I suspect we need to change upstream codebase too.

**Review by ravic-rs — COMMENTED** [review]
One comment to address

File: src/Main/Foedag.cpp (hunk 1 above)
Comment (KochynVolodymyr) [reply src/Main/Foedag.cpp:386]: compiler pointer pass to report manager. Report manager requires compiler pointer to recognize opensta.

**Review by ravic-rs — APPROVED** [review]

**Pull request merged**